src/game-logic/reducers/applyCrows.js

```diff
@@ -1,39 +1,53 @@
-import { itemType } from '../../enums'
-import { itemsMap } from '../../data/maps'
-import { getPlotContentType } from '../../utils'
-import { CROW_CHANCE } from '../../constants'
-import { CROW_ATTACKED } from '../../templates'
+import { doesPlotContainCrop } from '../../utils'
+import { CROW_CHANCE, MAX_CROWS } from '../../constants'
+import { CROWS_DESTROYED } from '../../templates'
 
-import { fieldHasScarecrow, updateField } from './helpers'
+import { modifyFieldPlotAt } from './modifyFieldPlotAt'
+
+import { fieldHasScarecrow } from './helpers'
 
 // TODO: Add tests for this reducer.
 /**
  * @param {farmhand.state} state
  * @returns {farmhand.state}
  */
 export const applyCrows = state => {
-  const { field } = state
+  const { field, purchasedField } = state
+
+  if (fieldHasScarecrow(field) || Math.random() > CROW_CHANCE) {
+    return state
+  }
+
   const newDayNotifications = [...state.newDayNotifications]
 
   let notificationMessages = []
+  const plotsWithCrops = []
 
-  const updatedField = fieldHasScarecrow(field)
-    ? field
-    : updateField(field, plotContent => {
-        if (!plotContent || getPlotContentType(plotContent) !== itemType.CROP) {
-          return plotContent
-        }
+  field.forEach((row, y) =>
+    row.forEach((plotContents, x) => {
```

Comment:
I think we can use [the `forRange` reducer](https://github.com/jeremyckahn/farmhand/blob/15d97be44cbed0cf618d602d5adcd8aeaf8726f6/src/game-logic/reducers/forRange.js#L10-L30) here to DRY this nested loop up a bit.

(Tangentially, I'm noticing that `forRange`'s JSDoc annotations are innacurate. I'll fix that in a seperate PR, but that shouldn't prevent it from being usable here.)

Reply:
PR for JSDoc update: https://github.com/jeremyckahn/farmhand/pull/358

Reply:
approved! 

re: `forRange`... i see what you're thinking here but it feels a little strange because of the radius arg. i guess i could just use a very large number, or do something like use the field width? i could also see a util being added like `forEachPlot` that is similar to `forRange` but loops over every plot and calls a function with similar args as `forRange` does. sorta thinking out loud, but i'm open to whatever you think is best here.

Reply:
i updated to use `forRange` but it still feels a bit awkward to me. mainly because that reducer takes `state` as an arg and doesn't really simplify getting access to the plot contents. using it still requires the caller to know how to dig into `state.field` along with the requirement to return an updated `state` value. i think something like this would still be ideal, but if this is the only usecase for it then it's a little tricky to justify

```js

Reply:
Thanks for giving `forRange` a try. I agree that using it here looks weird!

I really like the `forEachPlot` helper you've got sketched out here. I think it would make sense to add it as a utility function and use it here. On the other hand, that would be a pretty textbook case of premature refactoring. I think the simplest solution for this PR is to revert the use of `forRange` and put it back the way you had it. If you'd like to implement `forEachPlot`, I think it would be small and useful enough to justify expanding the scope of this PR a bit. Whichever solution you feel like going with is fine with me.

Sorry for going back and forth on this! I hadn't considered how using `forRange` here would actually look. 😅

Reply:
updated this again! i ended up making `forEachPlot` but just local in this file. if it's ever needed elsewhere it can be pulled out to a util but i think just keeping it local is the way to go for now

Reply:
I think that's a great solution! Good thinking.